tests/device_tests/test_session_id_and_passphrase.py

```diff
@@ -375,6 +376,40 @@ def call(passphrase: str, expected_result: bool):
     call(passphrase="A" * 49 + "š", expected_result=False)
 
 
+@pytest.mark.setup_client(passphrase=True)
+def test_hide_passphrase_from_host(client: Client):
+    # Without safety checks, turning it on fails
+    with pytest.raises(TrezorFailure, match="Safety checks are strict"), client:
+        device.apply_settings(client, hide_passphrase_from_host=True)
+
+    device.apply_settings(client, safety_checks=SafetyCheckLevel.PromptTemporarily)
+
+    # Turning it on
+    device.apply_settings(client, hide_passphrase_from_host=True)
+
+    passphrase = "abc"
+
+    response = client.call_raw(XPUB_REQUEST)
```

Comment:
would be nice to also assert on the text on Trezor screen, but I'm not sure how difficult that is to add here.
please try but if it's more than "easy", let's not stop the PR on it

Reply:
It was possible with the use of `input_flows` - the testing code is now much longer, but more thorough, as it also tests the amount of `ButtonRequests` that differs according to `hide` option. Done in [c2cd56a](https://github.com/trezor/trezor-firmware/pull/2829/commits/c2cd56a68a607224e2d392b32d60c2bc7481cb9d)